### PR TITLE
Add personal catalog

### LINKED ISSUE #39: Add favorites catalog to User class

### PR DESCRIPTION
**Pull request by caydenreynolds**
Users can now add and remove projects from their personal catalog. Note that there is no html/servlet side system that allows these actions to be taken
closes #39

### CODE REVIEW

File: src/ypd/model/project.py [hunk 1]
```diff
@@ -19,43 +28,48 @@ class Project(Base, HasUserMixin):
...
         Raises:
             ValueError: If no project with the given id exists
         """
-        session = Session()
-        # Look for project ID in database
         try:
-            result = session.query(cls).filter_by(id = id).one()
+            return session.query(cls).filter_by(id = id).one()
         except NoResultFound as e:
             raise ValueError(f'No project found with id {id}') from e
```
Comment (gabeholley222) [inline src/ypd/model/project.py]: is that f supposed to be in the argument? (f'No project found with id {id}')

Comment (QuietCoder1007) [reply src/ypd/model/project.py]: Yeah, it's an f-string. It's equivalent to a str.format to pass variables into a string like "Hello, my name is {}".format(name)

Comment (gabeholley222) [reply src/ypd/model/project.py]: Alright, I just wanted clarification before approving the merge to make sure it was intentional.

Comment (caydenreynolds) [reply src/ypd/model/project.py]: Yea, LeQuient's right. I, personally, favor f-strings because they're easy to read

**Pull request merged**